Fixes for kopano setup

jnweiger opened the pull request:
Here is a PR for the fixes found by @pmaier1

He says:
I tried that docker-compose set up out of curiosity and have it running. maybe my findings on the way can help you:
1) Caddy didn't like this https://github.com/owncloud-docker/compose-playground/blob/master/compose/kopano/konnect/Caddyfile#L5 / https://github.com/owncloud-docker/compose-playground/blob/master/compose/kopano/konnect/Caddyfile#L32 => changing to
tls internal made it work for me (see https://caddyserver.com/docs/caddyfile/directives/tls)
2) The comma at the end of this line is wrong and makes the automated LDAP config fail https://github.com/owncloud-docker/compose-playground/blob/master/compose/ldap/ldap.sh#L54
3) You need to manually sync LDAP users => occ user:sync "OCA\User_LDAP\User_Proxy"
4) You need to manually enable the OIDC app => occ a:e openidconnect
5) For me it only works with https://github.com/owncloud/openidconnect/releases/download/v0.2.0/openidconnect-0.2.0.tar.gz => the RC builds seem wrong (I noticed they are half the size of 0.2.0)

jnweiger committed:
Here is a PR for the fixes found by @pmaier1
He says:
I tried that docker-compose set up out of curiosity and have it running. maybe my findings on the way can help you:
1) Caddy didn't like this https://github.com/owncloud-docker/compose-playground/blob/master/compose/kopano/konnect/Caddyfile#L5 / https://github.com/owncloud-docker/compose-playground/blob/master/compose/kopano/konnect/Caddyfile#L32 => changing to
tls internal made it work for me (see https://caddyserver.com/docs/caddyfile/directives/tls)
2) The comma at the end of this line is wrong and makes the automated LDAP config fail https://github.com/owncloud-docker/compose-playground/blob/master/compose/ldap/ldap.sh#L54
3) You need to manually sync LDAP users => occ user:sync "OCA\User_LDAP\User_Proxy"
4) You need to manually enable the OIDC app => occ a:e openidconnect
5) For me it only works with https://github.com/owncloud/openidconnect/releases/download/v0.2.0/openidconnect-0.2.0.tar.gz => the RC builds seem wrong (I noticed they are half the size of 0.2.0)

jnweiger commented:
@davitol @jesmrec please review!

jnweiger committed:
refer to tls docs

jnweiger committed:
missing export ...

jnweiger committed:
More fixes for the kopano setup
from @jesmrec, thanks!
https://github.com/owncloud-docker/compose-playground/issues/6#issuecomment-681869927

jnweiger commented:
@pmaier1 the syntax 'tls internal' in the Caddyfile does not work for me. 

```
caddy_1           | 2020/08/27 16:01:13 [ERROR] Making new certificate manager: registration error: acme: error: 400 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-acct :: urn:ietf:params:acme:error:invalidEmail :: Error creating new account :: "internal" is not a valid e-mail address, url:  (attempt 2/3)
caddy_1           | 2020/08/27 16:01:13 [INFO] acme: Registering account for internal

caddy_1           | 2020/08/27 16:01:15 [ERROR] Making new certificate manager: registration error: acme: error: 400 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-acct :: urn:ietf:params:acme:error:invalidEmail :: Error creating new account :: "internal" is not a valid e-mail address, url:  (attempt 3/3)
caddy_1           | 2020/08/27 16:01:15 registration error: acme: error: 400 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-acct :: urn:ietf:params:acme:error:invalidEmail :: Error creating new account :: "internal" is not a valid e-mail address, url: 
caddy_1           | exit status 1
compose_caddy_1 exited with code 1
```
Caddy v1: tls self_signed
Caddy v2: tls internal

Are you running Caddy v2???

jnweiger commented:
@jesmrec I tried your suggestion and used this line in the environment:
```
- OWNCLOUD_APPS_INSTALL=https://github.com/owncloud/openidconnect/releases/download/v1.0.0RC2/openidconnect-1.0.0RC2.tar.gz
```

It gets deleted. :-( It seems we need another trick, when the major version of a app changes:

```
owncloud_1        | Installing openidconnect-1.0.0RC2.tar.gz app...
owncloud_1        | openidconnect: /tmp/openidconnect-1.0.0RC2.tar.gz has a different major version, try market:upgrade --major instead.
owncloud_1        | Deleting openidconnect-1.0.0RC2.tar.gz tarball...
owncloud_1        | Enabling openidconnect app...
owncloud_1        | openidconnect enabled
owncloud_1        | ownCloud is already latest version
owncloud_1        | Writing objectstore config...

jnweiger commented:
Caddy has a phone home feature. This should be disabled in our config!
```
caddy_1           | 2020/08/27 18:15:40 [WARNING] Sending telemetry (attempt 3): Post https://telemetry.caddyserver.com/v1/update/bd6ad771-5fd8-4bf4-9faa-ae257068424c:

jnweiger committed:
Caddy 1.0.3 cannot do 'tls internal' this is a caddy 2.1.1 directive!
Added a Caddyfile.v2 to play around with.

jnweiger commented:
In the current state of this PR:
* `docker system prune` **and** `docker volume prune` should be run before compose up, to avoid outdated data in internal volumes!
* caddy either v1 or v2 can be used (compose is using v1 right now)
* access owncloud with http://owncloud.docker-playground.local:7680
* ports 80 and ports 443 are occupied.
* the idp config is visible at https://konnect.docker-playground.local/.well-known/openid-configuration
* but actually using kopano for login, results in an error:
![image](https://user-images.githubusercontent.com/1108546/91496440-638e7e80-e8bc-11ea-9c4f-8984e22c3812.png)

The .local domain names prevent us from getting real certificates. Maybe that would help?

jesmrec commented:
> @jesmrec I tried your suggestion and used this line in the environment:
> 
> ```
> - OWNCLOUD_APPS_INSTALL=https://github.com/owncloud/openidconnect/releases/download/v1.0.0RC2/openidconnect-1.0.0RC2.tar.gz
> ```
> 
> It gets deleted. :-( It seems we need another trick, when the major version of a app changes:
> 
> ```
> owncloud_1        | Installing openidconnect-1.0.0RC2.tar.gz app...
> owncloud_1        | openidconnect: /tmp/openidconnect-1.0.0RC2.tar.gz has a different major version, try market:upgrade --major instead.
> owncloud_1        | Deleting openidconnect-1.0.0RC2.tar.gz tarball...
> owncloud_1        | Enabling openidconnect app...
> owncloud_1        | openidconnect enabled
> owncloud_1        | ownCloud is already latest version
> owncloud_1        | Writing objectstore config...
> ```

It seems that it tries to upgrade but finally realised that correct app version is in there. After finishing, `occ app:list` returns

```
Enabled:
  - activity: 2.5.3
  - comments: 0.3.0
  - configreport: 0.2.0
  - dav: 0.6.0
  - federatedfilesharing: 0.5.0
  - federation: 0.1.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_mediaviewer: 1.0.3
  - files_pdfviewer: 0.11.1
  - files_sharing: 0.13.0
  - files_texteditor: 2.3.0
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - firstrunwizard: 1.2.0
  - market: 0.6.0
  - notifications: 0.5.2
  - openidconnect: 1.0.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.4.0
  - updatenotification: 0.2.1
  - user_ldap: 0.15.2
```

is this not correct? in case the current version tarball is not installed, the previous one would be 0.2.0 (latest stable release number)

pmaier1 commented:
> It gets deleted. :-( It seems we need another trick, when the major version of a app changes:

It's just deleting the tarball for cleanup after installing the app if I'm not mistaken :) 

> owncloud_1        | openidconnect: /tmp/openidconnect-1.0.0RC2.tar.gz has a different major version, try market:upgrade --major instead.

If you start with a clean environment (docker system prune) this message should not appear as there is no upgrade but directly installing the respective version.

jnweiger committed:
Update README.md

fix export syntax

jnweiger committed:
Update README.md

jnweiger committed:
Update README.md

exact instructions for working with compose

jnweiger commented:
My setup is now reproducable with https://github.com/owncloud-docker/compose-playground/blob/master/examples/hetzner-deploy/make_openidconnect_test.sh

The remove messages are gone, the ldap sync works, 
but still caddy does not allow the secure ssl connection that seems to be needed. 
When I click the 'Kopano' button on the login screen, I get:

![image](https://user-images.githubusercontent.com/1108546/91874884-0a876780-ec7b-11ea-83a3-8758e5a77b77.png)

The server logs say
```
caddy_1           | 2020/09/01 15:46:50 http: TLS handshake error from 172.18.0.7:59290: local error: tls: bad record MAC
owncloud_1        | 2.247.255.253 - - [01/Sep/2020:15:46:50 +0000] "GET /apps/openidconnect/redirect HTTP/1.1" 503 9205 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:79.0) Gecko/20100101 Firefox/79.0"
caddy_1           | 2.247.255.253 - - [01/Sep/2020:15:46:50 +0000] "GET /apps/openidconnect/redirect HTTP/2.0" 503 8523

jnweiger committed:
Update Caddyfile.v2

Add wildcard matchers. The routes are exact with v2.
remove port numbers and tls settings. 

The defaults work, eveything ovespecific fails.

jnweiger committed:
switch docker compose to caddy v2

caddy v2 can generate certs.

jnweiger committed:
typo

davitol commented:
Using the instance provided by @jnweiger , when I try to login with an LDAP user using the Kopano setup I get:

<img width="752" alt="Screenshot 2020-09-02 at 08 21 15" src="https://user-images.githubusercontent.com/10596560/92080337-b20eb200-edc1-11ea-8723-d11813ff4f94.png">



`{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"--","app":"OpenID","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Access token: eyJhbGciOiJQUzI1NiIsImtpZCI6Imtvbm5lY3RkLXRva2Vucy1zaWduaW5nLWtleSIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJvd25DbG91ZCIsImV4cCI6MTU5OTExNTgzNCwianRpIjoiN2xNMF9iUDhrQmhHZmlwbUlQcldKU2xxcm55cW9UcW4iLCJpYXQiOjE1OTkxMTUyMzQsImlzcyI6Imh0dHBzOi8va29ubmVjdC5vaWRjLWp3LXFhLm93bmNsb3VkLndvcmtzIiwic3ViIjoiUzFEZjFyeG15UmlNTU95MGI2WnN1UHdWWGxGdUNEdldncmlpWXZjZFpfTTVvNlhrVE1KRExaTHVUc1hDejFTR2picmdMM0tKdEJZMXhKS3hlOGpsc0FAa29ubmVjdCIsImtjLmlzQWNjZXNzVG9rZW4iOnRydWUsImtjLmF1dGhvcml6ZWRTY29wZXMiOlsiZW1haWwiLCJvcGVuaWQiLCJwcm9maWxlIl0sImtjLmlkZW50aXR5Ijp7ImtjLmkuZG4iOiJhYWxpeWFoIiwia2MuaS5pZCI6InVpZD1hYWxpeWFoX2FkYW1zLG91PXBlb3BsZSxkYz1vd25jbG91ZHFhLGRjPWNvbSIsImtjLmkudW4iOiJhYWxpeWFoX2FkYW1zIn0sImtjLnByb3ZpZGVyIjoiaWRlbnRpZmllci1sZGFwIn0.T76QIFWEfyVT0-aRcmWBOkGkb4DoHypkrknaS6QkeUjy5AfRauF7c5jYdWTt0JkP1Zf6iXlfnoQVWOU3c5_HLDxIau3L9VgLmXAZJ7DqnVN1Yvr_SFED9cMyMNAHdQu0IU0kjsfBDdpdw2ejdbBkxYKhNfPSpCW8uAL9hO2eKPOD7YyDRzaesWO_tIgufBNuwJYBrVpsGs7RvqPE5uPsWSODlWqG-jtm10fnYwQok4vZMjX7ZXK8sjLbHAedL93E03K65ratOI7tbhn1snBrJy3F2b_Sw0TLvfRHLcNrG_2QVEH1clEg4UdTN_7I9nvjhtiJWG61P6H0F_JKHPzqo8B_qgagfk0-Vd5xBRlJhnPidWJDiUNUydULtge_zJiR5QPeFR09ky8Xh4X6UIhw4-uu9ucqKN-1z8bwdZZbljshH92F0aNfCH7NrBECKyIb2EsfAiYY8fyDDCHg2wRLxc-nz9-In8tBmQ8s-UdeM58YUe-juFhKhEg40iDeoesVJMRacY2ttf6iDaZ22VD4Y7hqZRJRVyIjSf02HB32JUN_9DeT2k55mmtAZzlnJn9XnFYxChooyWLlPIH5RKuhW4JIA0nITLMfauDsGRpVIMdaucLGfpsnXL0mD2qkF6inmJ9SMbq55p7ePxjR2DasJ5_6V-v-QT2_6LVTsWU_MRg"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"--","app":"OpenID","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Refresh token: "}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"--","app":"OpenID","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"User info: {\"email_verified\":false,\"family_name\":\"adams\",\"name\":\"aaliyah\",\"preferred_username\":\"aaliyah_adams\",\"sub\":\"S1Df1rxmyRiMMOy0b6ZsuPwVXlFuCDvWgriiYvcdZ_M5o6XkTMJDLZLuTsXCz1SGjbrgL3KJtBY1xJKxe8jlsA@konnect\"}"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"--","app":"OC\\Authentication\\Token\\DefaultTokenProvider::generateToken","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"generating token 7c2dbee663ebd968a51f09b47de0c5afafaf1e8ae6fa1061a2bffb91cfa990962bf92982efc424e6b829b0ba34f0fc2f5cc0522d339310e486fdf781e0b36e30, uid aaliyah_adams, loginName aaliyah_adams, pwd empty, name Mozilla\/5.0 (Macintosh; Intel Mac OS X 10.15; rv:80.0) Gecko\/20100101 Firefox\/80.0, type temporary"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Abandoning paged search - last cookie: '', cookies: <>"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Function ldap_control_paged_result() is deprecated at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/LDAP.php#293"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Initializing paged search for Filter objectClass=* base Array\n(\n    [0] => uid=aaliyah_adams,ou=people,dc=owncloudqa,dc=com\n)\n attr Array\n(\n    [0] => primarygroupid\n)\n at 0-500"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Cookie for uid=aaliyah_adams,ou=people,dc=owncloudqa,dc=com at 0-500 is ''"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Abandoning paged search - last cookie: '', cookies: <>"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Function ldap_control_paged_result() is deprecated at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/LDAP.php#293"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Ready for a paged search with cookie '' at 0-500"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Function ldap_control_paged_result() is deprecated at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/LDAP.php#293"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Requested attribute primarygroupid not found for uid=aaliyah_adams,ou=people,dc=owncloudqa,dc=com"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Abandoning paged search - last cookie: '', cookies: <>"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Function ldap_control_paged_result() is deprecated at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/LDAP.php#293"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Function ldap_control_paged_result() is deprecated at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/LDAP.php#293"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Function ldap_control_paged_result_response() is deprecated at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/LDAP.php#74"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Page response cookie '' at -0, estimated<0>"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"user_ldap","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Abandoning paged search - last cookie: '', cookies: <''>"}
{"reqId":"pTbetSAnmWzP3BjlgauV","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/index.php\/apps\/openidconnect\/redirect?code=wSvdCU17SVzndUZrvLA0AKx5GW04QJbm&scope=openid%20profile%20email&session_state=2997e4844981a7a46810a71f85f9d5d5415b20fd756caa17500d9e769b1b2129.pGj7QG88BLgs33slwuZ_an8ynOyzVQZgblKIIO-V5Uc%3D&state=fef47796bba1704cba4550e8a64b73f1","message":"Function ldap_control_paged_result() is deprecated at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/LDAP.php#293"}
{"reqId":"girsIG9fEI3ahBD7fVD8","level":3,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"no app in context","method":"GET","url":"\/apps\/files\/","message":"Exception: {\"Exception\":\"Jumbojett\\\\OpenIDConnectClientException\",\"Message\":\"The provider introspection_endpoint could not be fetched. Make sure your provider has a well known configuration available.\",\"Code\":0,\"Trace\":\"#0 \\\/mnt\\\/data\\\/apps\\\/openidconnect\\\/vendor\\\/jumbojett\\\/openid-connect-php\\\/src\\\/OpenIDConnectClient.php(482): Jumbojett\\\\OpenIDConnectClient->getWellKnownConfigValue()\\n#1 \\\/mnt\\\/data\\\/apps\\\/openidconnect\\\/vendor\\\/jumbojett\\\/openid-connect-php\\\/src\\\/OpenIDConnectClient.php(1352): Jumbojett\\\\OpenIDConnectClient->getProviderConfigValue()\\n#2 \\\/mnt\\\/data\\\/apps\\\/openidconnect\\\/lib\\\/SessionVerifier.php(104): Jumbojett\\\\OpenIDConnectClient->introspectToken()\\n#3 \\\/mnt\\\/data\\\/apps\\\/openidconnect\\\/lib\\\/Application.php(79): OCA\\\\OpenIdConnect\\\\SessionVerifier->verifySession()\\n#4 \\\/mnt\\\/data\\\/apps\\\/openidconnect\\\/appinfo\\\/app.php(25): OCA\\\\OpenIdConnect\\\\Application->boot()\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(239): require_once('\\\/mnt\\\/data\\\/apps\\\/...')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(190): OC_App::requireAppFile()\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(124): OC_App::loadApp()\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(886): OC_App::loadApps()\\n#9 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/mnt\\\/data\\\/apps\\\/openidconnect\\\/vendor\\\/jumbojett\\\/openid-connect-php\\\/src\\\/OpenIDConnectClient.php\",\"Line\":520}"}
{"reqId":"girsIG9fEI3ahBD7fVD8","level":0,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"PHP","method":"GET","url":"\/apps\/files\/","message":"Illegal offset type in isset or empty at \/var\/www\/owncloud\/lib\/private\/legacy\/app.php#287"}
{"reqId":"girsIG9fEI3ahBD7fVD8","level":3,"time":"2020-09-03T06:40:34+00:00","remoteAddr":"62.42.141.235","user":"aaliyah_adams","app":"index","method":"GET","url":"\/apps\/files\/","message":"Exception: {\"Exception\":\"Error\",\"Message\":\"Object of class OCA\\\\OpenIdConnect\\\\Application could not be converted to string\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(190): OC_App::requireAppFile()\\n#1 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(124): OC_App::loadApp()\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(886): OC_App::loadApps()\\n#3 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#4 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php\",\"Line\":245}"}`

davitol commented:
OOTH just clicking on the Kopano button (alternative login) These logs are spotted:

`{"reqId":"ILAvhDGzogQmxcBseWPQ","level":0,"time":"2020-09-03T06:48:24+00:00","remoteAddr":"62.42.141.235","user":"--","app":"OpenID","method":"GET","url":"\/apps\/openidconnect\/redirect","message":"Entering LoginFlowController::login"}
{"reqId":"ILAvhDGzogQmxcBseWPQ","level":0,"time":"2020-09-03T06:48:24+00:00","remoteAddr":"62.42.141.235","user":"--","app":"OpenID","method":"GET","url":"\/apps\/openidconnect\/redirect","message":"Before openid->authenticate"}`

pmaier1 commented:
> {\"Exception\":\"Jumbojett\\\\OpenIDConnectClientException\",\"Message\":\"The provider introspection_endpoint could not be fetched. Make sure your provider has a well known configuration available.

I'm getting this as well with 1.0.0RC2, not getting it with 0.2.0. @DeepDiver1975 said it's a config issue.

pmaier1 commented:
You need to remove this line to make it work: https://github.com/owncloud-docker/compose-playground/blob/master/compose/kopano/konnect/openid.sh#L16

jnweiger committed:
remove 'use-token-introspection-endpoint' => false,

as suggested in https://github.com/owncloud-docker/compose-playground/pull/51#issuecomment-686341504

jnweiger committed:
workaround while waiting for https://github.com/owncloud/openidconnect/pull/98

jnweiger committed:
Today's findings:
- .well-known must be available in owncloud domain
- identifier-registration.yml allows a name: attribute. It is displayed instead of the id, if available.

For: https://github.com/owncloud/openidconnect/issues/66#issuecomment-704972704

Commit:
try ro redirect the well-known via the openidconnect app, according to
https://github.com/owncloud/openidconnect/blame/master/README.md#L65

DeepDiver1975 commented on compose/kopano/konnect/Caddyfile.v2:
```diff
@@ -27,7 +27,10 @@
+	# reverse_proxy /.well-known/* kopano_konnect:8777
 	reverse_proxy owncloud:8080 {
+	  ## more correct well-known, handled by the openidconnet app:
+	  redir /.well-known/openid-configuration /index.php/apps/openidconnect/config
```
redir means redirect? so http/30x ?

if so this is wrong - as per spec well-known is not allowed to be redirected

Commit:
use rewrite instead of redir in caddy. That is an internal redirect!

jnweiger committed:
add hint only in Caddyfile to do the same as Caddyfile.v2 -- not implemented, not needed currently.
